### DISCUSSION TITLE
Query related to use of different time steps in Multiapp

### DISCUSSION THREAD

**Discussion by deepakbiitk** [idaholab/moose#24395] [Q&A General] [answered]
Hi all,

I am solving a weakly coupled thermomechanical problem. I adopted the use of Multiapp. In the main app, I solve heat conduction and I am then transferring the   temperature data to the sub app and solve for deformation. I want to use two different time steps. like a staggered approach i.e., solve heat equation with a time step dt and transfer the data to sub app in the intervals of 10*dt, 20*dt, 30*dt, .... and solve for deformation. 

Any suggestions how can I set this in my input file in the MultiApps and Transfer section. Thanks.

**Comment by GiudGiud** [comment]
Hello

Typically you would flip the two apps, then you could run the subapp more often than the parent app.
This is the main criterion for deciding which app is the main app, it s the one running the longest time step.

If you dont want to do that, you can turn on and off transfers and multiapps using the controls system.
https://mooseframework.inl.gov/source/controls/TimePeriod.html
You ll have to create this frequency (once every 10 dt) yourself, I dont think we have that one

Guillaume

**Comment by deepakbiitk** [comment]
Hello Guillaume,

Thank you for the prompt response. I flipped my two apps. However, I am still not able to make it run with 2 different time steps.

This is the section in my parent app:

[AuxVariables]
  [./temperature]
  [../]
[]

[MultiApps]
  [sub_app]
    type = TransientMultiApp
    input_files = 'transient_thermal_main.i'
    execute_on = 'timestep_begin'
  []
[]

[Transfers]
  [from_sub]
    type = MultiAppCopyTransfer
    from_multi_app = sub_app
    variable = temperature
    source_variable= temperature
    execute_on = 'timestep_begin'
  []
[]

My time step in sub app is dt =1 and time step in parent app is dt = 10. When I run it, both the apps run with dt =1. Thanks

**Reply by GiudGiud** [reply, marked as answer]
turn on sub_cycling in the MultiApp sub_app block

**Reply by deepakbiitk** [reply]
Thanks, It worked.
